packages/contracts/contracts/optimistic-ethereum/OVM/execution/OVM_ExecutionManager.sol

```diff
@@ -951,16 +951,19 @@ contract OVM_ExecutionManager is iOVM_ExecutionManager, Lib_AddressResolver {
     // expensive by touching a lot of different accounts or storage slots. Since most contracts
     // only use a few storage slots during any given transaction, this shouldn't be a limiting
     // factor.
+    // Here we will limit the Nuisance Gas forwarded to the external message to the minimum of
+    // the nuisance gas left in this call frame, and the gas limit provided to the call. This protects
+    // against an attack where an untrusted contract intentionally uses up all available nuisance
+    // gas.
     uint256 prevNuisanceGasLeft = messageRecord.nuisanceGasLeft;
-    uint256 nuisanceGasLimit = _getNuisanceGasLimit(_gasLimit);
```

Comment:
This line is effectively: 
`Math.min(_gasLImit, gasleft())`
Which does not take into account the possibility that `nuisanceGasLeft` might be less than those values. 
[ref](https://linear.app/optimism/issue/OP-648/oz-c03-unbounded-nuisance-gas)